src/lib.rs

```diff
@@ -341,6 +341,57 @@ impl<const LEN: usize> AStr<LEN> {
         };
         AStr::<RET_LEN>::from_utf8_array_unchecked(ret_buf)
     }
+
+    pub fn try_from_fmt(display: impl core::fmt::Display) -> Result<Self, core::fmt::Error> {
+        use core::fmt::Write;
+        let mut builder = FmtBuilder::new();
+        write!(builder, "{}", display)?;
+        builder.finalize()
+    }
+}
+
+/// Private type to build an [`AStr`] from anything that can print to an [core::fmt::Write]
+struct FmtBuilder<const LEN: usize> {
+    len: usize,
+    partial: AStr<LEN>,
+}
+
+impl<const LEN: usize> FmtBuilder<LEN> {
+    pub fn new() -> Self {
```

Comment:
why are these functions `pub` if the struct is private?

Reply:
This would have been to minimize friction if `FmtBuilder` is later moved to another file. I will change it if you prefer.